backend/t4cclient/sessions/files/routes.py

```diff
@@ -86,3 +87,24 @@ def upload_files(
     OPERATOR.upload_files(id, tar_bytes)
 
     return {"message": "Upload successful"}
+
+
+@router.get(
+    "/download",
+    responses=AUTHENTICATION_RESPONSES,
+)
+def download_file(
+    id: str,
+    filename: str,
+    db: Session = Depends(get_db),
+    token=Depends(JWTBearer()),
+) -> UploadFile:
+    check_session_belongs_to_user(get_username(token), id, db)
+
+    return StreamingResponse(
+        OPERATOR.download_file(id, filename),
+        headers={
+            "content-disposition": 'attachment; filename="download.zip"',
```

Comment:
The filename should not be `download.zip`. I messes up the Download directory on the file system und you have no idea which download it was later on. We could just use the name of the directory.